### PR TITLE
Add sim_afu

### PR DESCRIPTION
**Pull request by michael-adler**
sim_afu loads the ASE version of libopae-c, turning software compiled for use
with hardware into a simulation run.

### CODE REVIEW

**Comment by michael-adler** [comment]
What do you all think of this? I've been struggling with the problem of having to write Makefiles for examples that either build both ASE and FPGA versions of a binary or a mode that builds one or the other. This script lets us just compile for FPGA and invoke it for ASE by forcing it to load libopae-c-ase.

Any suggestions for the script name? sim_afu is ok, but maybe there is something better.

**Comment by deepakcu84** [comment]
Why can't you use a flag in the Makefile? E.g. make USE_ASE=1 builds the ASE version.

If you still want to keep the script, how about "run_as_ase", "run_in_ase" or "use_ase" instead of sim_ase?

**Comment by michael-adler** [comment]
I like "run_in_ase".

The reason I don't like the USE_ASE=1 flag, especially, is because it builds a single named binary that has different semantics depending on the flag. You never know how it was compiled last, so are forced to do "make clean" every time you do a make in order to be sure it is compiled how you need it.

**Comment by chipb** [comment]
I also like `run_in_ase` and this script in general.

It'd be an atypical case, but I think we should *extend* LD_PRELOAD rather than overriding entirely:
```
env LD_PRELOAD="libopae-c-ase.so${LD_PRELOAD:+ $LD_PRELOAD}" "$@"
```

**Comment by abelardojarab** [comment]
Is there a consistent methodology on the motivations for naming these scripts? I think it would be useful to see what is being envisioned here.

File: ase/scripts/run_in_ase [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+# Run a binary using the ASE simulation library instead of the standard libopae-c.
+env LD_PRELOAD="libopae-c-ase.so${LD_PRELOAD:+:$LD_PRELOAD}" "$@"
```
Comment (chipb) [inline ase/scripts/run_in_ase]: Huh. I think I've only seen `LD_PRELOAD` as being space-delimited, not `:`-delimited, but I guess the dynamic linker accepts either.

**Comment by chipb** [comment]
Not to dwell on something inconsequential, but another name occurred to me. How does plain `with_ase` sound?

As a sanity check to potentially add later, it might be worth verifying that `libopae-c-ase.so` appears on the search path prior to launching the command.

**Comment by michael-adler** [comment]
@chipb I think you are probably right about `with_ase`. I was finding `run_in_ase` tedious to type.

Any disagreement? Not of deep consequence, but typed a lot so worth a couple extra minutes.

**Comment by deepakcu84** [comment]
I'm fine with with_ase

**Comment by michael-adler** [comment]
@OPAE/ci waiting for you to approve one line make change.

**Pull request merged**